src/solver/conv_MP_bidirectional_winograd.cpp

```diff
@@ -368,96 +348,138 @@ ConvSolution ConvMPBidirectWinograd<WinoDataH, WinoFilterH, WinoDataW, WinoFilte
     void* reserved_ptr = nullptr;
     int unused         = 0;
 
-    size_t wino_in_offset = 0, wino_out_offset = wino_in.buff_info.total_byte_size,
-           wino_wei_offset = wino_out_offset + wino_out.buff_info.total_byte_size;
-
-    // GEMM
-    int m = K, k = C,
-        n   = wino_in.buff_info.size.nk * wino_in.buff_info.size.w * wino_in.buff_info.size.h;
-    int lda = m, ldb = n, ldc = n;
-    int batch_count       = wino_xform_h * wino_xform_w * group_cnt;
-    long long int strideA = m * k * 1LL, strideB = k * n * 1LL, strideC = m * n * 1LL;
-    float alpha = 1., beta = 0.0;
-    const bool isColMajor = false, transA = true, transB = false;
-    // clang-format off
-    GemmDescriptor wino_gemm_desc{isColMajor,transA,transB,m,n,k,
-        lda,ldb,ldc,batch_count,strideA,strideB,
-        strideC,alpha,beta,params.in_data_type};
+    DEFINE_WORKSPACE_OFFSETS(wino_in.buff_info.total_byte_size,
+                             wino_wei.buff_info.total_byte_size,
+                             wino_out.buff_info.total_byte_size)
+
+    InvokerFactory gemm_conv_factory;
+    std::string gemm_conv_kernel_name;
+
+    auto zeroDesc = TensorDescriptor();
+    if(isXdlops)
+    {
+        gemm_conv_kernel_name = "XDLOPS_CONV: ";
+        gemm_conv_factory     = xdlops_factory;
+    }
+    else
+    {
+        // GEMM
+        gemm_conv_kernel_name = "WRW_WINO_GEMM: ";
+
+        int m = K, k = C,
+            n   = wino_in.buff_info.size.nk * wino_in.buff_info.size.w * wino_in.buff_info.size.h;
+        int lda = m, ldb = n, ldc = n;
+        int batch_count       = wino_xform_h * wino_xform_w * group_cnt;
+        long long int strideA = m * k * 1LL, strideB = k * n * 1LL, strideC = m * n * 1LL;
+        float alpha = 1., beta = 0.0;
+        const bool isColMajor = false, transA = true, transB = false;
+        // clang-format off
+        GemmDescriptor wino_gemm_desc{isColMajor,transA,transB,m,n,k,
+            lda,ldb,ldc,batch_count,strideA,strideB,
+            strideC,alpha,beta,params.in_data_type};
+
+        gemm_conv_factory = [=](const std::vector<Kernel> &) {
+
+            return [=](const Handle& handle, const boost::any& ctx) {
+#if(MIOPEN_USE_ROCBLAS)
+                const auto data_ctx = boost::any_cast<conv::DataInvokeParams>(ctx);
+                Data_t workSpace = data_ctx.workSpace;
+                CallGemmStridedBatched(handle,
+                    wino_gemm_desc,
+                    workSpace,
+                    static_cast<int>(wino_wei_offset / GetTypeSize(params.in_data_type)),
+                    workSpace,
+                    static_cast<int>(wino_in_offset / GetTypeSize(params.in_data_type)),
+                    workSpace,
+                    static_cast<int>(wino_out_offset / GetTypeSize(params.in_data_type)),
+                    nullptr,
+                    false,
+                    GemmBackend_t::rocblas);
+#else
+                (void)handle;
```

Comment:
Why not MIOPEN_THROW?

Reply:
``#if(MIOPEN_BACKEND_HIP && MIOPEN_USE_ROCBLAS)``
used only to avoid compiler error with OpenCl.
And ``MIOPEN_THROW`` used only for class functions to avoid errors if function must return an object.

Reply:
Thanks for fixing.